Tests/scs-compliance-check.py

```diff
@@ -245,115 +470,60 @@ def main(argv):
             "sections": config.sections,
             "forced_version": config.version or None,
             "forced_tests": None if config.tests is None else config.tests.pattern,
-            "invocations": {},
+            "invocations": invocations,
         },
     }
-    check_keywords('spec', spec)
-    if config.version:
-        spec["versions"] = [vd for vd in spec["versions"] if vd["version"] == config.version]
+
+
+def main(argv):
+    """Entry point for the checker"""
+    logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+    config = Config()
+    config.apply_argv(argv)
+    if not config.subject:
+        raise RuntimeError("You need pass --subject=SUBJECT.")
+    with open(config.arg0, "r", encoding="UTF-8") as specfile:
+        spec = yaml.load(specfile, Loader=yaml.SafeLoader)
+    check_keywords('spec', spec)  # super simple syntax check (recursive)
```

Comment:
`check_keywords` returns `None` or an integer, but doesn't abort with an exception, and the return value isn't checked here

Reply:
addressed with 07085fc39efbf2b3657e9fec196fd99ab9018fca
now return number of errors, only raise exception when spec is fully checked